### PR TITLE
Implement internal IP allocation for Internal LoadBalancer

### LINKED ISSUE #760: IP assignment for Internal `LoadBalancer`
Implement Internal IP allocation for Internal Load Balancer's using IPAM Prefix in `onmetal-api`.

### PR DESCRIPTION
**Pull request by kasabe28**
# Proposed Changes

- Allocate IPAM Prefix IP for Internal LoadBalancer
- Add testcase to validate Internal LoadBalancer IP allocation

Fixes #760

### CODE REVIEW

File: internal/controllers/networking/loadbalancer_controller.go [hunk 1]
```diff
@@ -71,6 +75,17 @@ func (r *LoadBalancerReconciler) delete(ctx context.Context, log logr.Logger, lo
...
 
+	if loadBalancer.Spec.Type == networkingv1alpha1.LoadBalancerTypeInternal {
+		ips, err := r.applyInternalIPs(ctx, log, loadBalancer)
+		if err != nil {
+			return ctrl.Result{}, fmt.Errorf("error getting / applying internal ip: %w", err)
+		}
+		if err := r.updateStatus(ctx, log, loadBalancer, ips); err != nil {
+			return ctrl.Result{}, fmt.Errorf("error patching load balancer status")
+		}
+		log.V(1).Info("Patched load balancer status with internal ip")
```
Comment (adracus) [inline internal/controllers/networking/loadbalancer_controller.go:86]: Could you please move the `updateStatus` call to the bottom of the `reconcile` method? To 'save' the IPs, do like following pseudo code

```go
var ips []commonv1alpha1.IP
if loadBalancer.Spec.Type == networkingv1alpha1.LoadBalancerTypeInternal {
	var err error
	ips, err = r.applyInternalIPs(ctx, log, loadBalancer)
	if err != nil {
		return ctrl.Result{}, fmt.Errorf("error getting / applying internal ips: %w", err)
	}
}
...
if err := r.updateStatus(ctx, log, loadBalancer, ips); err != nil {
	return ctrl.Result{}, fmt.Errorf("error patching load balancer status: %w")
}
log.V(1).Info("Updated load balancer status")

File: internal/controllers/networking/loadbalancer_controller.go [hunk 2]
```diff
@@ -71,6 +75,17 @@ func (r *LoadBalancerReconciler) delete(ctx context.Context, log logr.Logger, lo
 func (r *LoadBalancerReconciler) reconcile(ctx context.Context, log logr.Logger, loadBalancer *networkingv1alpha1.LoadBalancer) (ctrl.Result, error) {
 	log.V(1).Info("Reconcile")
 
+	if loadBalancer.Spec.Type == networkingv1alpha1.LoadBalancerTypeInternal {
+		ips, err := r.applyInternalIPs(ctx, log, loadBalancer)
+		if err != nil {
+			return ctrl.Result{}, fmt.Errorf("error getting / applying internal ip: %w", err)
+		}
+		if err := r.updateStatus(ctx, log, loadBalancer, ips); err != nil {
+			return ctrl.Result{}, fmt.Errorf("error patching load balancer status")
```
Comment (adracus) [inline internal/controllers/networking/loadbalancer_controller.go:84]:
```suggestion
			return ctrl.Result{}, fmt.Errorf("error patching load balancer status: %w", err)
```

File: internal/controllers/networking/loadbalancer_controller_test.go [hunk 3]
```diff
@@ -104,4 +108,78 @@ var _ = Describe("LoadBalancerReconciler", func() {
...
+		))
+
+		By("asserting it get's an internal IP")
+		Eventually(Object(loadBalancer)).Should(HaveField("Status.IPs", ContainElements(*commonv1alpha1.MustParseNewIP("10.0.0.0"))))
+
+		By("deleting the load balancer")
+		Expect(k8sClient.Delete(ctx, loadBalancer)).To(Succeed())
+
+		By("waiting for it to be gone")
+		Eventually(Get(loadBalancer)).Should(Satisfy(apierrors.IsNotFound))
```
Comment (adracus) [inline internal/controllers/networking/loadbalancer_controller_test.go:183]: The deletion part is not required / can be removed.

**Review by adracus — CHANGES_REQUESTED** [review]
Some minor comments, otherwise good work!

File: internal/controllers/networking/loadbalancer_controller.go (hunk 1 above)
Comment (kasabe28) [reply internal/controllers/networking/loadbalancer_controller.go]: If `loadBalancer.Spec.NetworkInterfaceSelector` is not provided, `loadBalancer.Status` wont get updated.

File: internal/controllers/networking/loadbalancer_controller.go (hunk 2 above)
Comment (kasabe28) [reply internal/controllers/networking/loadbalancer_controller.go]: Done

File: internal/controllers/networking/loadbalancer_controller_test.go (hunk 3 above)
Comment (kasabe28) [reply internal/controllers/networking/loadbalancer_controller_test.go]: Done

**Review by adracus — APPROVED** [review]

**Pull request merged**